### PR TITLE
H-plugin synapse workspace diagnostic logs enabled

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.synapse, function(location, rcb) {
+            const workspaces = helpers.addSource(cache, source,
+                ['synapse', 'listWorkspaces', location]);
+
+            if (!workspaces) return rcb();
+
+
```
Comment (AkhtarAmir) [inline plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js:27]:
```suggestion
```

File: plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                helpers.addResult(results, 3, 'Unable to query Synapse workspaces: ' + helpers.addError(workspaces), location);
+                return rcb();
+            }
+
+            if (!workspaces.data.length) {
+                helpers.addResult(results, 0, 'No existing Synapse workspaces found', location);
+                return rcb();
+            }
+
+            for (let workspace of workspaces.data) {
```
Comment (AkhtarAmir) [inline plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js:38]:
```suggestion
            for (let workspace of workspaces.data) {
                  if (!workspace.id) continue;
```

File: plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+module.exports = {
+    title: 'Synapse Workspace Diagnostic Logging Enabled',
+    category: 'AI & ML',
+    domain: 'Machine Learning',
+    severity: 'Medium',
+    description: 'Ensures diagnostic logging is enabled for Synapse workspace.',
+    more_info: 'Enabling diagnostic logging for Azure Synapse workspace enhances performance monitoring, troubleshooting, and security optimization. This feature captures detailed logs of workspace activities, allowing you to gain insights, identify issues, and maintain the integrity and efficiency of data operations.',
+    recommended_action: 'Enable diagnostic logging for all Synapse workspaces.',
+    link: 'https://learn.microsoft.com/en-gb/azure/azure-monitor/essentials/diagnostic-settings',
```
Comment (AkhtarAmir) [inline plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js:12]: can you cross check the link?

File: plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Synapse Workspace Diagnostic Logging Enabled',
+    category: 'AI & ML',
+    domain: 'Machine Learning',
+    severity: 'Medium',
+    description: 'Ensures diagnostic logging is enabled for Synapse workspace.',
```
Comment (AkhtarAmir) [inline plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js:9]:
```suggestion
    description: 'Ensures that diagnostic logging is enabled for Synapse workspace.',
```

File: plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    more_info: 'Enabling diagnostic logging for Azure Synapse workspace enhances performance monitoring, troubleshooting, and security optimization. This feature captures detailed logs of workspace activities, allowing you to gain insights, identify issues, and maintain the integrity and efficiency of data operations.',
```
Comment (AkhtarAmir) [inline plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js:10]:
```suggestion
    more_info: 'Enabling diagnostic logs in Azure Synapse workspace is important for monitoring, troubleshooting, and optimizing performance. These logs provide detailed insights into resource usage, query execution, and potential issues, allowing administrators to identify bottlenecks, track errors, and improve the overall efficiency and reliability of the workspace.',
```

File: exports.js [hunk 6]
```diff
@@ -1222,8 +1222,10 @@ module.exports = {
         'workspaceManagedIdentity'      : require(__dirname + '/plugins/azure/synapse/workspaceManagedIdentity.js'),
         'synapseWorkspaceAdAuthEnabled' : require(__dirname + '/plugins/azure/synapse/synapseWorkspaceAdAuthEnabled.js'),
         'synapseWorkspacPrivateEndpoint': require(__dirname + '/plugins/azure/synapse/synapseWorkspacPrivateEndpoint.js'),
+        'workspaceDiagnosticLogsEnabled': require(__dirname + '/plugins/azure/synapse/workspaceDiagnosticLogsEnabled.js'),
         'workspaceDoubleEncryption'     : require(__dirname + '/plugins/azure/synapse/workspaceDoubleEncryption.js'),
 
+
```
Comment (alphadev4) [inline exports.js:1228]:
```suggestion
```

**Pull request merged**